dask/array/creation.py

```diff
@@ -677,17 +680,17 @@ def _diag_len(dim1, dim2, offset):
 
 
 @derived_from(np)
-def tri(N, M=None, k=0, dtype=float, chunks="auto"):
+def tri(N, M=None, k=0, dtype=float, chunks="auto", like=None):
     _min_int = np.lib.twodim_base._min_int
 
     if M is None:
         M = N
 
     chunks = normalize_chunks(chunks, shape=(N, M), dtype=dtype)
 
-    m = greater_equal.outer(
-        arange(N, chunks=chunks[0][0], dtype=_min_int(0, N)),
-        arange(-k, M - k, chunks=chunks[1][0], dtype=_min_int(-k, M - k)),
+    m = greater_equal(
+        arange(N, chunks=chunks[0][0], dtype=_min_int(0, N), like=like).reshape(1, N).T,
+        arange(-k, M - k, chunks=chunks[1][0], dtype=_min_int(-k, M - k), like=like),
```

Comment:
Because CuPy doesn't support `greater_equal.outer`, I rewrote this in terms of a matrix transpose. I don't know if this imposes any performance impacts for NumPy-based use cases, if that seems like a problem please speak up @Illviljan .

Reply:
I haven't done any timeits but I imagine this is similar to what's done in numpy in the end so I'm not particularly worried. It should be an improvement over the previous dask implementation anyway.

Reply:
Thanks for confirming.